### DISCUSSION TITLE
C: What needs to be standardized?

### DISCUSSION THREAD

**Discussion by ml-evs** [MADICES/MADICES-2022#5] [Ideas]
This thread is for discussing, finalizing and organizing a breakout with the theme in the title above. 

[Link to current description](https://github.com/MADICES/MADICES-2022/blob/main/breakouts/C_what-to-standardize.md)

> # What should we standardize? 
> 
> - Standardize on the data, format, model, or on the tool or the transport level? 
> - Tool development is more complicated/time-consuming than the development of parsers. 
>     - How can we remain format agnostic? 
>     - Can we leave people their own formats but “tag”/”annotate” concepts? 
>     - Does this help? 
>     - Is it practical? 
>     - How could it be implemented? 
>     - What specific technologies can be used to enable interoperability?
> - Linked data formats and the semantic web: 
>     - JSON-LD to provide context and ontologies to make agreement explicit.
>     - Is “Linked” Open Data too much of a barrier? 
>     - Can tools/frameworks like LinkML help bridge the schema > ontology gap?
> - Data Format vs data model:
>     - How do we make sure the model travels with the serialization?
> - Are centralised software libraries required to support such an effort, or would they simply be useful? 
>     - Who would write these? 
>     - What “higher-order” libraries could be useful? (e.g. a GitHub action that takes a Python library, with some config, and creates an API server, with machine-readable metadata/schemas published on an “app registry”)
> - Learning from other fields: 
>     - If you have something scalable for chemistry, does it scale to other domains? 
>     - If it doesn’t, why not? 
>     - If there are standards for other domains (e.g. sample-based sciences), why can’t we use them? 
> 
>

**Comment by kjappelbaum** [comment]
To play devil's advocate and kick off a discussion: 
"Is standardizing vocabularies not equivalent to standardizing file formats and data models?"

**Reply by sphuber** [reply]
I echo this. I see the hesitation to address the problem of standards because of the complexity and the risk of getting stuck discussing minute details. However, I think it is a fundamental part of the problem and it will have to be addressed at some point. Perhaps what we can do is try and discuss ways to standardize a small part of what can be standardized and do it in a way that it is as flexible as possible. Maybe if we are able to build some minimal scaffolding that can already be useful to start making things slightly interoperable and work from there.

**Reply by ml-evs** [reply]
I think my take on this aspect ultimately comes down to: "what would it take for people to describes samples with RDF in chemistry/mat. sci?" RDF is the natural low level framework that any recommendations we make must be "meta"-interoperable with. If we do not think it is suitable/workable in our field, then I think we at least need a good excuse! We can see ongoing efforts to make CIF interoperable with RDF via the [CIF ontology](https://github.com/emmo-repo/CIF-ontology). This to me is the separation between vocab/model: there is no reason why several non-aligned vocabularies cannot co-exist, and hopefully mappings between vocabularies will be clarified.

Going "full RDF" for each individual project carries a huge implementation burden (see e.g. https://github.com/w3c/EasierRDF and https://www.w3.org/Data/events/data-ws-2019/assets/position/David%20Booth.pdf) and there are so many choices that have to be made along the way that it can be completely paralyzing without some kind of longer term commitment to e.g. ontology development/publishing.

What could the middle ground be? Are there tools that allow RDF to be adopted progressively? Without wanting to suggest we "standardize it", you can imagine a lightweight markup that implicitly contains some of the assumptions we can rely on, e.g. a JSON file alongside an archive that describes links between rows in a spreadsheet and raw data files.

**Reply by ml-evs** [reply]
A second big aspect is on the interoperability of tooling, which ties in with breakout #8. There would need to be some standardization to enable machine discoverability of libraries, APIs etc. that are compatible with given data. Does this need a shared vocabulary? Can we make an "app store" for e.g. spectral data that describes the required fields to ingest/export an XAS object using e.g. the IUPAC taxonomy of techniques? A lot of this is reinventing the wheel from chemical ontology efforts, but I am not aware of any software tools (for e.g., peak picking, structural assignment) or infrastructure that themselves are "semantic". Is the current scientific software ecosystem now more amenable to this approach, compared to 10-15 years ago?

**Comment by kjappelbaum** [comment]
For starters we can ask 
- "In your domain, what do you think is lacking standardization; or, in contrast, is already well standardized". 
- How would these standards play along with existing tools?
- Would a controlled vocabulary (list), maybe even a thesaurus be enough?
- Who in your community could lead this effort? What would be needed to solve the collective action problem?

**Comment by kjappelbaum** [comment]
A valuable contribution from our side/the workshop would be already to just get a better overview what is out there, what are the limitations of existing solutions, what are the overlaps. 

For instance, https://arxiv.org/abs/2002.03842, seems relatively incomplete to me.

**Comment by kjappelbaum** [comment]
@shyamd in chat

>My belief is one of the problems is that these ontologies and vocabularies are too horizontal: trying to span all of x-ray chemistry for instance, instead of choosing one aspect and describing it out in detail. For instance, could someone write an ontology for just all the x-ray techniques focusing on how they are related and differentiated?

**Reply by kjappelbaum** [reply]
Alejandra
>Ontologies can be modularised, so it is feasible to focus on specific areas indeed. This would be expected as each topic in domain ontologies should be tackled by experts of the domain

**Reply by kjappelbaum** [reply]
>Right, I would like to see very narrowly focused ontologies by individuals that we can discuss validity. Once the community agrees, we can stitch them back into a more general framework, but  trying to write the general framework from scratch is significant mental challenge.

**Reply by kjappelbaum** [reply]
Ruben
>I think some core "objects" are quite mandatory. Such as "chemical species" etc.

**Reply by kjappelbaum** [reply]
>I wouldn’t even enforce that til the community adoption phase, so that individual ontology authors aren’t constrained. Those core concepts can be translated at that phase.

**Reply by kjappelbaum** [reply]
>Imagine a python like ecosystem for ontologies in chemistry and matsci where individuals act as BIFDL for their tiny scope and the community through something like the PEP protocol integrates them in

**Reply by kjappelbaum** [reply]
Ken 
>I’d push back on the modularized ontology approach insofar as how we define synonyms for the leaf-level terminology is actually not a globally valid tree, but rather application dependent.  It’s always better to have more annotation, but we don’t obviate the cross-walk challenges just by starting at the lowest levels.

**Comment by kjappelbaum** [comment]
Ruben in chat

>Is JSON something we can agree on regarding interoperability? During the first talk and the discussion regarding "automation islands" i was always picturing JSONs or something similar as an intermediate point.

**Reply by kjappelbaum** [reply]
Ken
>JSON is a data layer.  Presuming people adopt JSON-LD (which better maps to controlled vocabulary concepts), it is homomorphic to other data representations and thus we really need to prioritize semantics rather than syntactics.

**Reply by kjappelbaum** [reply]
response by amarine
>I agree. JSON, yaml, XML... They are all interchangeable. It is not about the format in which we represent data, is about the way we treat it
